websockify/token_plugins.py

```diff
@@ -87,3 +87,19 @@ class JSONTokenApi(BaseTokenAPI):
     def process_result(self, resp):
         resp_json = resp.json()
         return (resp_json['host'], resp_json['port'])
+
+
+class JWTTokenApi(BasePlugin):
+    # source is a JWT-token, with hostname and port included
+    # this JWT is validated against a assymetric key, if provided else, nothing is checked!
+
+    def lookup(self, token):
+        import jwt
+
+        try:
+            secret = open(self.source).read()
+            parsed = jwt.decode(token, secret, algorithms=['RS256', 'RS384', 'RS512', 'HS256', 'HS384', 'HS512', 'ES256', 'ES384', 'ES512', 'PS256', 'PS384', 'PS512'])
+
+            return (parsed['host'], parsed['port'])
+        except:
```

Comment:
It would be nice if there was a way to log the exception that happened so you can debug things.